test/integration/progressUpdates.test.js

```diff
@@ -0,0 +1,204 @@
+    it("Marked the task as monitored by superuser", function (done) {
+      chai
+        .request(app)
+        .post(`/progressupdates/${taskId}`)
+        .set("Cookie", `${cookieName}=${superUserAuthToken}`)
+        .send({
+          monitor: true,
+          frequency: 2,
+        })
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res).to.have.status(200);
+          expect(res.body).to.be.a(object);
+          expect(res.body).to.have.keys(["message", "id"]);
+          expect(res.body.id).to.be.equal(taskId);
+          expect(res.body.message).to.be.equal("task marked for progress updates");
+          const storedTask = tasks.fetchTask(taskId);
+          expect(storedTask.monitored).to.be.equal(true);
+          expect(storedTask.frequency).to.be.equal(2);
+          return done();
+        });
+    });
+
+    it("Unmarked the task as monitored by superuser", function (done) {
+      taskId = tasks.updateTask(taskData[5]);
+      chai
+        .request(app)
+        .post(`/progressupdates/${taskId}`)
+        .set("Cookie", `${cookieName}=${superUserAuthToken}`)
+        .send({
+          monitor: false,
+        })
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res).to.have.status(200);
+          expect(res.body).to.be.a(object);
+          expect(res.body).to.have.keys(["message", "id"]);
+          expect(res.body.id).to.be.equal(taskId);
+          expect(res.body.message).to.be.equal("task unmarked for progress updates");
+          const storedTask = tasks.fetchTask(taskId);
+          expect(storedTask.monitored).to.be.equal(false);
+          return done();
+        });
+    });
+
+    it("Returns 401 for normal users", function (done) {
+      taskId = tasks.updateTask(taskData[5]);
+      chai
+        .request(app)
+        .post(`/progressupdates/${taskId}`)
+        .set("Cookie", `${cookieName}=${jwt}`)
+        .send({
+          monitor: false,
+        })
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res).to.have.status(401);
+          expect(res.body.message).to.be.equal("Unauthenticated User");
+          return done();
+        });
+    });
+
+    it("Returns 404 for invalid task id's", function (done) {
+      chai
+        .request(app)
+        .post(`/progressupdates/123`)
+        .set("Cookie", `${cookieName}=${jwt}`)
+        .send({
+          monitor: false,
+        })
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res).to.have.status(404);
+          expect(res.body.message).to.be.equal("Task id not found");
+          return done();
+        });
+    });
+  });
+
+  describe("Test the save api", function () {
+    it("Saves the progress updates", function (done) {
+      chai
+        .request(app)
+        .post(`progressupdates/save/${taskId}`)
+        .set("Cookie", `${cookieName}=${jwt}`)
+        .send(progressData[0])
```

Comment:
I'd suggest storing `progessData[0]` in a meaningful variable name and using it here. It will make it easy to understand the test better

Reference: https://github.com/Real-Dev-Squad/website-backend/blob/a6b34d51595dbb30368285a5dff517e935e17685/test/integration/members.test.js#L18-L25

Reply:
Yes, IMO we should refactor how fixtures are being used in other tests as well in the future